### PR TITLE
docs: cookbook visual fix for cards and small ci doc dev changes

### PR DESCRIPTION
**Pull request by onuralpszr**
- Fix cookbook cards that make them clickable
    - Combine multiple pip install commands into a single step
    - Simplify the workflow by reducing redundancy
    - Ensure all necessary dependencies are installed in one go

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -195,7 +195,7 @@ So far, **there is no type checking with mypy**. See [issue](https://github.com/
 
 The `supervision` documentation is stored in a folder called `docs`. The project documentation is built using `mkdocs`.
 
-To run the documentation, install the project requirements with `poetry install --with dev`. Then, run `mkdocs serve` to start the documentation server.
+To run the documentation, install the project requirements with `poetry install --with docs`. Then, run `mkdocs serve` to start the documentation server.
```
Comment (LinasKo) [inline CONTRIBUTING.md:198]: Let's keep `--with dev`. I don't want to surprise anyone.

File: pyproject.toml [hunk 2]
```diff
@@ -82,13 +82,14 @@ docutils = [
...
 mkdocs-material = { extras = ["imaging"], version = "^9.5.5" }
-mkdocstrings = { extras = ["python"], version = ">=0.20,<0.26" }
+mkdocstrings = { extras = ["python"], version = "^0.25.2" }
 mike = "^2.0.0"
 # For Documentation Development use Python 3.10 or above
 # Use Latest mkdocs-jupyter min 0.24.6 for Jupyter Notebook Theme support
 mkdocs-jupyter = "^0.24.3"
 mkdocs-git-committers-plugin-2 = "^2.2.3"
 mkdocs-git-revision-date-localized-plugin = "^1.2.4"
+jupyterlab = "^4.2.4"
```
Comment (LinasKo) [inline pyproject.toml:92]: We probably don't need this, do we?

**Review by LinasKo — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (onuralpszr) [reply CONTRIBUTING.md]: Done

File: pyproject.toml (hunk 2 above)
Comment (onuralpszr) [reply pyproject.toml]: Done removed

**Comment by onuralpszr** [comment]
@LinasKo both changes done.

**Comment by LinasKo** [comment]
This is great!

1. Tested cookbooks cards - works great.
2. What's a good way to test the CI changes? Can I run it easily somehow?

**Comment by onuralpszr** [comment]
> This is great!
> 
> 1. Tested cookbooks cards - works great.
> 2. What's a good way to test the CI changes? Can I run it easily somehow?

If you want to test "CI" locally in your computer you what you can do repeat what I did in CI command line 


```console
curl -LsSf https://astral.sh/uv/install.sh | sh
uv venv
source .venv/bin/activate
uv pip install "mkdocs-material" "mkdocstrings[python]" "mkdocs-material[imaging]" mike "mkdocs-git-revision-date-localized-plugin" jupyterlab mkdocs-jupyter mkdocs-git-committers-plugin-2
mkdocs build --verbose
```

If you want more complete CI test locally you need to setup "https://github.com/nektos/act" (it is a good way to repeat github action in same environment locally in your computer by using container

**Comment by LinasKo** [comment]
This is faster, and I'm not seeing any regressions locally. I'm also actively using `uv`, so it feels like a welcome change.

Still, it's a change in how we build the CI, so I'll want a review from @SkalskiP too.

**Review by LinasKo — DISMISSED** [review]

**Comment by onuralpszr** [comment]
Small friendly correction, this is run on every PR created, every commit in PR and PR after merged. Not just after merge I did not change how is it run it will still run as same previously but faster.

**Review by LinasKo — APPROVED** [review]
Dismissed my review on the wrong PR

**Review by LinasKo — APPROVED** [review]

**Pull request merged**